Source/Core/Core/NetPlayServer.cpp

```diff
@@ -584,6 +586,51 @@ unsigned int NetPlayServer::OnData(sf::Packet& packet, Client& player)
 	}
 	break;
 
+	case NP_MSG_TIMEBASE:
+	{
+		u32 x,y, frame;
+		packet >> x;
+		packet >> y;
+		packet >> frame;
+		s_timebase[frame].emplace_back(x | ((u64)y >> 32), player.pid);
+
+		if (!std::all_of(s_timebase[frame].begin(), s_timebase[frame].end(), [&frame](std::pair<u64, u8> i){ return i.first == s_timebase[frame][0].first; }))
+		{
+			sf::Packet spac;
+			spac << (MessageId) NP_MSG_DESYNC_DETECTED;
+
+			int pid = -1;
+			if (s_timebase[frame].size() > 2)
+			{
+				for (auto hash : s_timebase[frame])
```

Comment:
As far as i can tell it is correct. It gives a count of how many are equal to hash.first